src/Features/ui/AllSkills/SkillItem.tsx

```diff
@@ -0,0 +1,89 @@
+import {Box, Typography} from '@mui/material'
+import {FC, useContext, useEffect, useState} from 'react'
+import {SkillBar, SkillBarFilled, SkillItemButton} from './StyledComponents'
+import {EditContext} from './AllSkills'
+import {TransformedArray} from '../FormOver/FormOver'
+import {useParams} from 'react-router-dom'
+import {getCurrentUserID} from '@/App'
+
+export interface SkillItemProps {
+  name: string
+  proficiency: number
+  mastery: string
+  categoryId: string
+  category: string
+}
+
+const SkillItem: FC<SkillItemProps> = ({
+  name,
+  proficiency,
+  mastery,
+  categoryId,
+  category,
+}) => {
+  const context = useContext(EditContext)
```

Comment:
Для контекста можно сделать деструктуризацию 
```suggestion
  const {isEdit, setEdit, formOpen} = useContext(EditContext)
```
Плюс посмотреть другие места/файлы